src/lib/utilities/decode-payload.ts

```diff
@@ -76,6 +76,31 @@ export const decodePayloadAttributes = (
   return anyAttributes;
 };
 
+// List of fields with payloads
+const payloadFields: string[] = [
+  'input',
+  'result',
+  'details',
+  'heartbeatDetails',
+  'lastHeartbeatDetails',
+  'lastFailure',
+  'lastCompletionResult',
+  'queryArgs',
+  'answer',
+  'signalInput',
+];
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getPotentialPayloads = (anyAttributes: any): Payload[] | null => {
```

Comment:
We gotta retype our types to fix this (which we need to do in the near future)

  // This anyAttributes allows us to use ?. notation to introspect the object which is a safe access pattern.
  // Because of the way we wrote our discriminated union we have to use this any. If we have two objects that
  // don't have the same property TS won't let us access that object without verifying the type string like
  // attributes.type === "ATypeWithInput/Result"